addon/controllers/admin/model-records/show/related/new.js

```diff
@@ -16,32 +16,30 @@ export default Controller.extend(
 
     actions: {
       save(model, parentModel) {
-        const recordName = this.get('recordName')
-        const inverseRelationshipName =
-          this.inverseRelationshipName(parentModel, recordName)
-        const kind =
-          this.relationshipKind(parentModel, recordName)
-        const inverseKind =
-          this.inverseRelationshipKind(parentModel, recordName)
-
-        if (inverseKind === 'belongsTo') {
-          model.set(inverseRelationshipName, parentModel)
-        } else {
-          model.get(inverseRelationshipName).pushObject(parentModel)
-        }
+        const {
+          inverseRelationshipKind,
+          inverseRelationshipName,
+          relationshipKind,
+          relationshipName
+        } = this.relationshipInfo(parentModel)
 
         model.save().then(record => {
-          if (kind !== 'belongsTo' && kind !== 'hasMany') {
-            return this._transitionToRelated()
-          }
-
-          if (kind === 'belongsTo') {
-            parentModel.set(recordName, record)
-          } else {
-            parentModel.get(recordName).pushObject(record)
-          }
-
-          parentModel.save().then(() => this._transitionToRelated())
+          parentModel
+            .addRelated(relationshipName, record, relationshipKind)
+            .then(() => {
+              if (relationshipKind === 'belongsTo') {
```

Comment:
Similar question here.  It seems like this should be synced automatically or else handled in the addRelated method...